### PR TITLE
1.2 backports 18-09-12

### PR DESCRIPTION
**Pull request by ianvernon**
Backported PRs:

* #5319 (had some minor conflicts, @eloycoto please review)
* #5503
* #5513
   * I only backported commit 2c0385365cd930aba91590d3f8b36b010d3cfbeb, but not 724334bb037c93f3716602c3cdde149350b59af8 because 724334bb037c93f3716602c3cdde149350b59af8 fixed an issue in master that was not present in v1.2.
* #5515 - required fixing merge conflicts, but I was the author of the PR so I am confident in it working correctly.
* #5538 - required fixing merge conflicts. @aanm and @joestringer please review carefully.

*Not* backported PRs:

* #5445
    * Tons of merge conflicts. I do not think this needs to be backported because it fixes functionality that wasn't even added in the v1.2 branch. Replied on that PR with requested justification for need to backport.

Signed-off by: Ian Vernon <ian@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5541)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-docs-please

**Comment by ianvernon** [comment]
test-missed-k8s

**Comment by ianvernon** [comment]
Build failed due to a rogue import:
```
18:55:12 test/helpers/node.go:24:2: imported and not used: "time"
```

Fixing...

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-docs-please

**Comment by ianvernon** [comment]
test-missed-k8s

**Comment by ianvernon** [comment]
Hit gofmt issues... will fix up and try again.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

File: pkg/maps/ctmap/entry.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/cilium/cilium/pkg/byteorder"
+)
+
+// CtEntry represents an entry in the connection tracking table.
+type CtEntry struct {
```
Comment (joestringer) [inline pkg/maps/ctmap/entry.go]: This patch backport would be much simpler if the `Slave` field is just added to the existing CtEntry structure instead. Then the backport commit would also match line-for-line the commit being backported.

**Review by joestringer — CHANGES_REQUESTED** [review]
I'm inclined to request a simplification of the backport for #5538 , we shouldn't be refactoring the CtEntry in the backport. 

If you feel strongly about the current approach, feel free to dismiss this request for changes.

**Comment by ianvernon** [comment]
> I'm inclined to request a simplification of the backport for #5538 , we shouldn't be refactoring the CtEntry in the backport.

@joestringer sure, not a problem. I'm all for smallest-possible deltas in backports :)

**Comment by ianvernon** [comment]
@joestringer followed your recommendations.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Forgot to update `bpf.sha`, fixed and re-running tests.

**Comment by ianvernon** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Pull request merged**